src/Tasks/ResourceHandling/StringResource.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Resources;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Tasks.ResourceHandling
+{
+    internal class StringResource : IResource
+    {
+        public string Name { get; }
+
+        public string OriginatingFile { get; }
+
+        public string Value { get; }
+
+        public StringResource(string name, string value, string filename)
+        {
+            Name = name;
+            Value = value;
+            OriginatingFile = filename;
+        }
+
+        public void AddTo(IResourceWriter writer)
+        {
+            writer.AddResource(Name, Value);
+        }
+
+        public override bool Equals(object that)
+        {
+            if (that is StringResource other)
+            {
+                return Name == other.Name &&
+                    Value == other.Value;
+                // TODO: also OriginatingFile? I don't think I care
```

Comment:
Why does this class override Equals?

Reply:
It was to support some tests, but you're right that it's bogus; I'll write a helper comparer instead.